app/payment-link-v2/amount/post-index.controller.js

```diff
@@ -1,5 +1,46 @@
 'use strict'
 
-module.exports = async (req, res) => {
-  //TODO - Add code for post controller.  
+const lodash = require('lodash')
+
+const { response } = require('../../utils/response')
+const paths = require('../../paths')
+const { validateAmount } = require('../../utils/validation/server-side-form-validations')
+const replaceParamsInPath = require('../../utils/replace-params-in-path')
+const { setSessionVariable } = require('../../utils/cookie')
+
+const PAYMENT_AMOUNT = 'payment-amount'
+const BACK_LINK_HREF = 'back-link-href'
+
+function validateAmountFormValue (amount, res) {
+  const errors = {}
+  const amountValidationResult = validateAmount(amount)
+  if (!amountValidationResult.valid) {
+    errors[PAYMENT_AMOUNT] = res.locals.__p(amountValidationResult.messageKey)
+  }
+
+  return errors
+}
+
+module.exports = async function (req, res, next) {
+  const paymentAmount = lodash.get(req.body, PAYMENT_AMOUNT, '')
+  const backLinkHref = lodash.get(req.body, BACK_LINK_HREF, '')
```

Comment:
Rather than storing this in a hidden form field, it might be better to share the code to determine the URL for the back link and call this from both controllers?
I don't think the pattern of putting values like this in hidden form fields is something we would usually do. It probably wouldn't cause major issues with the behaviour but I'm unsure about introducing this pattern.

Reply:
Good point - I have moved the back link code to a shared function.